presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -95,6 +102,8 @@
     private final List<QueryResult> testPostQueryResults = new ArrayList<>();
 
     private boolean deterministic = true;
+    private boolean hasNewSessionProperties = false;
+    private Map<String, String> sessionPropertiesTemp;
```

Comment:
I see. My concern is that "sessionPropertiesTemp" requires two map copy operations. Implementing it as a local variable may trigger too many operations like that, which may not be desirable. How do you think?